storm-core/src/jvm/org/apache/storm/blobstore/LocalFsBlobStore.java

```diff
@@ -219,17 +219,44 @@ public void setBlobMeta(String key, SettableBlobMeta meta, Subject who) throws A
     @Override
     public void deleteBlob(String key, Subject who) throws AuthorizationException, KeyNotFoundException {
         validateKey(key);
-        checkForBlobOrDownload(key);
-        SettableBlobMeta meta = getStoredBlobMeta(key);
-        _aclHandler.hasPermissions(meta.get_acl(), WRITE, who, key);
+
+        boolean checkPermission = true;
+        try {
+            checkForBlobOrDownload(key);
+        } catch (KeyNotFoundException e) {
+            LOG.warn("Detected deletion from ZK for the key {} while pulling the latest change. Needs to be deleted. key: {}", key);
+            checkPermission = false;
+        }
+
+        if (checkPermission) {
```

Comment:
Instead of this can you combine the two `try...catch` into something like
```java
try {
 checkForBlobOrDownload(key);
 SettableBlobMeta meta = getStoredBlobMeta(key);
 _aclHandler.hasPermissions(meta.get_acl(), WRITE, who, key);
} catch (KeyNotFoundException e) {
 LOG.warn(...);
}
```